uart-msgpacker/src/main/java/nl/isaac/androidlibs/uart_msgpacker/packer/UARTUnpacker.kt

```diff
@@ -33,48 +33,96 @@ class UARTUnpacker {
+                    ResponseWrapper(
+                        null,
+                        WriteResponse(reset.rid, null, reset.resetMessages),
+                        null,
+                        null
+                    )
                 }
-                ResponseType.READ -> {
+                ResponseType.READ.get(shouldUseNewKeys) -> {
                     val readResponse = unpackToType(data, ReadResponse::class.java)
                     ResponseWrapper(readResponse, null, null, null)
                 }
-                ResponseType.LOGS -> {
+                ResponseType.LOGS.get(shouldUseNewKeys) -> {
                     val logResponse = unpackToListOfType(data, LogResponse::class.java)
                     ResponseWrapper(null, null, null, logResponse)
                 }
                 null -> ResponseWrapper(null, null, null, null)
+                else -> ResponseWrapper(null, null, null, null)
             }
         }
 
-        private fun determineResponseType(data: String): ResponseType? {
+        private fun determineResponseType(data: String, shouldUseNewKeys: Boolean): ResponseType? {
             return when {
-                data.contains(ResponseType.WRITE.idString) -> ResponseType.WRITE
-                data.contains(ResponseType.RESET_MESSAGES.idString) -> ResponseType.RESET_MESSAGES
-                data.contains(ResponseType.MESSAGES.idString) -> ResponseType.MESSAGES
-                data.contains(ResponseType.READ.idString) -> ResponseType.READ
-                data.contains(ResponseType.LOGS.idString) -> ResponseType.LOGS
+                data.contains(ResponseType.WRITE.idString(shouldUseNewKeys)) -> ResponseType.WRITE.get(
+                    shouldUseNewKeys
+                )
+                data.contains(ResponseType.RESET_MESSAGES.idString(shouldUseNewKeys)) -> ResponseType.RESET_MESSAGES.get(
+                    shouldUseNewKeys
+                )
+                data.contains(ResponseType.MESSAGES.idString(shouldUseNewKeys)) -> ResponseType.MESSAGES.get(
+                    shouldUseNewKeys
+                )
+                data.contains(ResponseType.READ.idString(shouldUseNewKeys)) -> ResponseType.READ.get(
+                    shouldUseNewKeys
+                )
+                data.contains(ResponseType.LOGS.idString(shouldUseNewKeys)) -> ResponseType.LOGS.get(
+                    shouldUseNewKeys
+                )
                 else -> null
             }
         }
 
         private enum class ResponseType(val idString: String) {
-            READ("read"), WRITE("write"), RESET_MESSAGES("resetMessages"), MESSAGES("message"), LOGS("logEntries")
+            READ("read"), WRITE("write"), RESET_MESSAGES("resetMessages"), MESSAGES("message"), LOGS(
+                "logEntries"
+            ),
+            READ_NEW("read"), WRITE_NEW("write"), RESET_MESSAGES_NEW("resetMessages"), MESSAGES_NEW(
+                "M"
+            ),
+            LOGS_NEW("L");
```

Comment:
I think having separate entries for regular and _new does not really make sense here. I think having it more like UARTConstants would be better. Now you check if data contains a certain key, then return a specific response type and then use the returned response type to retrieve the correct one again to do the deserializing. 

fun idString(boolean) : String can then just be return if (boolean) newIdString else oldIdString
determineResponseType can do data.contains(ResponseType.Type) -> ResponseType.Type
and unpackToResponseWrapper doesn't have to use the .get() method anymore (so I think that can just go?)

Reply:
I might be missing something here tho, so please let me know if that is the case

Reply:
I wanted to implement like UARTConstants, but I couldnt figure how, because this enum sometime is needed to return the ResponseType or the enum itself not the value inside.
The UARTConstants is easier because it is only needed to provide the value, old value or new value.
Maybe I see it too far and more complicated, 
I will try your suggestion if that works

Reply:
Your solution
`determineResponseType can do data.contains(ResponseType.Type) -> ResponseType.Type`
Is that means I need to do when condition for other new types as well? So it will be 10 when entries?

Reply:
I agree if I can remove the other function, 
I can actually remove the idString function, and just use get, and if I want to retrieve the value, I just need to call get.idString

Reply:
You don't need the the get at all I think.

I had a little typ-o before it should be like:
determineResponseType can do data.contains(ResponseType.Type.idString(boolean)) -> ResponseType.Type

Because all you're checking for is the string and returning the enum. In that case you don't need 10 entries, plus there's no functional difference between the old and new response types besides the idString.